glutin/src/api/egl/display.rs

```diff
@@ -362,7 +362,18 @@ impl Display {
                     handle.connection.map_or(egl::DEFAULT_DISPLAY as *mut _, |c| c.as_ptr()),
                 )
             },
-            RawDisplayHandle::Gbm(handle) if extensions.contains("EGL_MESA_platform_gbm") => {
+            RawDisplayHandle::Gbm(handle)
+                // NOTE: Some drivers report that they support KHR extension with 1.4 EGL display, so
+                // workaround here by checking the KHR gbm display as well. The MESA and KHR has
+                // the same constant values, thus it'll work regardless.
```

Comment:
Should we make it clearer that "they support KHR extension with 1.4 EGL" is the thing that's "not allowed / not supported by the spec"?

```suggestion
                // NOTE: Some drivers report that they support the KHR GBM extension without EGL 1.5 client, so
                // work around that here by checking the KHR GBM extension as well. The MESA and KHR extensions have
                // the same constant values, thus it'll work regardless.
```

Also, I think you meant to refer to the client version, not the display version? The client lacks the [`GetPlatformDisplay` function, implying EGL < 1.5](https://registry.khronos.org/EGL/sdk/docs/man/html/eglGetPlatformDisplay.xhtml), yet [EGL 1.5 is required by `EGL_KHR_platform_gbm`](https://registry.khronos.org/EGL/extensions/KHR/EGL_KHR_platform_gbm.txt).

(And to be extremely picky: the fact that the constants are the same value "doesn't really matter" when the spec doesn't define that the combination of function and constant value are valid. But the driver is already broken anyway, that's why it works)